build/AzurePipelinesTemplates/WindowsAppSDK-BuildAndTestMRT-Steps.yml

```diff
@@ -1,85 +1,31 @@
-    inputs:
-      filePath: 'BuildAll.ps1'
-      arguments: -Platform "$(buildPlatform)" -Configuration "$(buildConfiguration)" -AzureBuildStep "PreFastSetup"
-
-  # The PREFast task below only supports the VSBuild/MSbuild task, _not_ a direct call to MSBuild.exe.
-  - task: MSBuild@1
-    displayName: 'build MrtCore with MSBuild'
-    inputs:
-      platform: '$(buildPlatform)'
-      solution: '${{ parameters.MRTSourcesDirectory }}\mrt\MrtCore.sln'
-      configuration: '$(buildConfiguration)'
-      msbuildArguments: '/restore /binaryLogger:${{ parameters.MRTBinariesDirectory }}/mrtcore.$(buildPlatform).$(buildConfiguration).binlog'
-
-  # ToDo: WindowsAppSDK-BinaryAnalysis-steps.yml below ultimately invokes SDLNativeRules@2, which is out of support.
-  # So for the time being, we explicitly call SDLNativeRules@3, which is currently supported. Consider eliminating this 
-  # extra step when we run PREFast as part of WindowsAppSDK-BinaryAnalysis-steps.yml below.
-  - task: SDLNativeRules@3
-    displayName: 'Guardian: PreFast@3'
-    env:
-      SYSTEM_ACCESSTOKEN: $(System.AccessToken)
-    continueOnError: true
-    inputs:
-      userProvideBuildInfo: 'auto'
-      toolVersion: Latest
-
-- ${{ if or(ne(parameters.RunSDLBinaryAnalysis, 'true'), ne(parameters.EnablePREFast, 'true')) }}:
-  # PREFast scan is disabled in this pipeline run, just proceed to build the MRT via BuildAll.ps1, and
-  # fall through to run BinaryAnalysis with PREFast turned off.
-  - task: PowerShell@2
-    name: BuildBinaries
-    inputs:
-      filePath: 'BuildAll.ps1'
-      arguments: -Platform "$(buildPlatform)" -Configuration "$(buildConfiguration)" -AzureBuildStep "BuildMRT"
-
-# End of forked code paths.
+# This is required to fix VSTest failing with
+# ##[error]This operation requires installation of Visual Studio with the Universal Windows Platform development workload selected
+- task: PowerShell@2
+  displayName: "Installing Microsoft.VisualStudio.Workload.Universal component"
+  inputs:
+    targetType: inline
+    workingDirectory: $(Build.SourcesDirectory)
+    script: |
+      Measure-Command {Invoke-Expression "cd C:\'Program Files (x86)\Microsoft Visual Studio'\Installer\; .\vs_installer.exe modify --installPath 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise' --add Microsoft.VisualStudio.Workload.Universal --downloadThenInstall --quiet | Out-Null"; Write-Host "Total time taken:"} | Select-Object Minutes, Seconds
 
-# Common code for performing Binary Analysis.
-- ${{ if eq(parameters.RunSDLBinaryAnalysis, 'true') }}:
-  - template: WindowsAppSDK-BinaryAnalysis-steps.yml
-    parameters:
-      outputDirectory: 'BuildOutput'
-      # ToDo: Enable PREFast to run downstream from here, once the SDL template has been updated to invoke the 
-      # newer SDLNativeRules@3 task, and remove the explicit call to it above.
-      enablePREFast: false
+- task: PowerShell@2
+  name: BuildBinaries
+  inputs:
+    filePath: 'BuildAll.ps1'
```

Comment:
My understanding is that this change will prevent us from running PREFast on MRT. Is this intended? Or OneBranch has "magic" make it work?

Reply:
Yes all the sdl analysis should be injected for us.

Reply:
My understanding is that PREFast is not injected for us automatically. 
Even if it is, the SDL template is currently running the unsupported PREFast@2, that means we will lose auto bug filing, that's a step backwards.

Reply:
Synced Offline with Alex. We will bring back 'Guardian: PreFast@3' to the MRT Stage